modules/graph_ql/main.tf

```diff
@@ -0,0 +1,72 @@
+
+data "aws_iam_policy_document" "graphql_task_policy_document" {
+  statement {
+    actions   = ["sts:AssumeRole"]
+    resources = ["*"]
+  }
+}
+
+data "aws_iam_policy_document" "assume_ecs_tasks_role_policy_document" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+# IAM role that tasks can use to make API requests to authorized AWS services.
+resource "aws_iam_role" "orca_ecs_tasks_role" {
+  name                 = "${var.prefix}_orca_ecs_tasks_role"
+  assume_role_policy   = data.aws_iam_policy_document.assume_ecs_tasks_role_policy_document.json
+  permissions_boundary = var.permissions_boundary_arn
+  tags                 = var.tags
+}
+
+resource "aws_iam_role_policy" "graphql_task_role_policy" {
+  name   = "${var.prefix}_orca_graphql_task_role_policy"
+  role   = aws_iam_role.orca_ecs_tasks_role.id
+  policy = data.aws_iam_policy_document.graphql_task_policy_document.json
+}
+
+# This role is required by tasks to pull container images and publish container logs to Amazon CloudWatch on your behalf.
+resource "aws_iam_role" "orca_ecs_task_execution_role" {
+  name                 = "${var.prefix}_orca_ecs_task_execution_role"
+  assume_role_policy   = data.aws_iam_policy_document.assume_ecs_tasks_role_policy_document.json
+  permissions_boundary = var.permissions_boundary_arn
+  tags                 = var.tags
+}
+
+# Defines how the image will be run. container_definitions can be replaced by data element.
```

Comment:
Largely untested, but it did publish to AWS. Next cards in this epic are to create and test a basic frontend.